WalletWasabi.Fluent/Behaviors/ShowAttachedFlyoutWhenFocusedBehavior.cs

```diff
@@ -0,0 +1,133 @@
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Diagnostics;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+using ReactiveUI;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.Helpers;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+public class ShowAttachedFlyoutWhenFocusedBehavior : AttachedToVisualTreeBehavior<Control>
+{
+	public static readonly DirectProperty<ShowAttachedFlyoutWhenFocusedBehavior, bool> IsFlyoutOpenProperty =
+		AvaloniaProperty.RegisterDirect<ShowAttachedFlyoutWhenFocusedBehavior, bool>(
+			"IsFlyoutOpen",
+			o => o.IsFlyoutOpen,
+			(o, v) => o.IsFlyoutOpen = v);
+
+	private bool _isFlyoutOpen;
+
+	public bool IsFlyoutOpen
+	{
+		get => _isFlyoutOpen;
+		set => SetAndRaise(IsFlyoutOpenProperty, ref _isFlyoutOpen, value);
+	}
+
+	protected override void OnAttachedToVisualTree(CompositeDisposable disposable)
+	{
+		if (AssociatedObject is null)
+		{
+			return;
+		}
+
+		if (AssociatedObject.GetVisualRoot() is not Control visualRoot)
+		{
+			return;
+		}
```

Comment:
I am not sure of this is worth it, your call.

```suggestion
		if (AssociatedObject?.GetVisualRoot() is not Control visualRoot)
		{
			return;
		}
```

Reply:
Yours feels better.